hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/TokenCreateHandler.java

```diff
@@ -168,6 +168,7 @@ public void handle(@NonNull final HandleContext context) {
 
         // Update record with newly created token id
         recordBuilder.tokenID(newTokenId);
+        recordBuilder.tokenType(newToken.tokenType());
```

Comment:
This is implied from the transaction body. Do we need this in `TokenCreateRecord` ?

Reply:
The request was to add the explicit type in the record builder wherever reasonable so that BlockStream doesn't need to determine the type (and possibly become out of sync with the service in doing so).

Worth noting that the value will not be written to the Record; it will be used in the Builder to determine the correct type of "Output" object to use in Block Stream.

Reply:
I don't understand. I thought that it will be added to `TransactionRecord` object.  So, you mean we don't need to add it to the `build()` here ?

Reply:
Not necessarily, no.  BlockStream is replacing SingleTransactionRecord and Receipts with other objects (as I understand it; @nickpoorman can clarify better than I can), so the comment here is really just a note to help guide that replacement.

Reply:
@nickpoorman  Will this be a field in `TranscationRecord.proto` or just needed internally ?

Reply:
Understood, so it will be serialized in these protobufs files `TransactionResult` and `TransactionOutputs` for BlockStreams. Also, I assume it will be documented in HIP.

Agreed `TokenType` in record makes sense. I am just not sure if its needed in `TokenCreateRecord` since it can be inferred from the transactionBody by mirror node.

Reply:
Per external discussion, `TokenType` will be added to the `SingleTransactionRecod`, rather than `TransactionRecord`.  The comments are not quite in the correct place in the code as a result.

Reply:
Fixed.